va/va_trace.c

```diff
@@ -6316,3 +6316,49 @@ void va_TraceEventBuffers(
     }
     return;
 }
+
+void va_TraceExportSurfaceHandle(
+    VADisplay        dpy,
+    VASurfaceID      surfaceId,
+    uint32_t         memType,
+    uint32_t         flags,
+    void             *descriptor)
+{
+    int i;
+
+    DPY2TRACE_VIRCTX(dpy);
+
+    TRACE_FUNCNAME(idx);
+
+    va_TraceMsg(trace_ctx, "\tsurfaceId = 0x%08x\n", surfaceId);
+    va_TraceMsg(trace_ctx, "\tmemType   = 0x%08x\n", memType);
+    va_TraceMsg(trace_ctx, "\tflags     = 0x%08x\n", flags);
+
+    VADRMPRIMESurfaceDescriptor *desc = (VADRMPRIMESurfaceDescriptor *)descriptor;
```

Comment:
As noted at
https://github.com/intel/libva/blob/984dfee4177021c400367f5dffc0776a6dd745dc/va/va.h#L4054-L4055

You can't assume it's `VADRMPRIMESurfaceDescriptor` without checking `memType`.

Reply:
Fixed in last version, see line 6337 above.